rust/src/delta.rs

```diff
@@ -79,6 +79,21 @@ pub enum DeltaTableError {
     },
 }
 
+#[cfg(feature = "python")]
+pub mod python {
+    extern crate pyo3;
+    use self::pyo3::create_exception;
+    use self::pyo3::exceptions::PyException;
+
+    create_exception!(deltalake, PyDeltaTableError, PyException);
+
+    impl From<super::DeltaTableError> for pyo3::PyErr {
```

Comment:
I don't understand enough of what's going on here to tell why this trait implementation must exist in this crate versus the python crate :confused: _

Reply:
i tried to do this within python crate, but couldn't get it to compile because `DeltaTableError` and `pyo3::PyErr` are both defined in foreign crates. To define a From trait, either the destination or the end needs to defined in the current crate, which is not the case for python create :(

Reply:
I figured it might be something like this.

I think we might need to try to find a different approach here unfortunately. Would the `deltalake` crate be able to define `DeltaLakeError` as a trait and use an `impl DeltaTableError` where it's sending errors around, such that the Python crate would implement its own custom error on top?

My main concern here is that dependency features in Cargo workspaces are shared such that I don't believe the Ruby crate would be able to be built _without_ the `python` feature (I haven't yet checked this though).

I cannot remember exactly, but I think [this issue in Cargo](https://github.com/rust-lang/cargo/issues/4463) was the one I originally looked at.

Reply:
I moved the error chaining into python crate using `map_err`. The code is a little bit more verbose, but i think it's worth the tradeoff to remove a language specific feature from the main crate.